wordpress-mobile/WordPress-iOS
# Xpost Notifications: show `Follow conversation by email` on original post

Fixes #17158
WPKit PR: https://github.com/wordpress-mobile/WordPressKit-iOS/pull/446

When viewing the original post from an xpost Notification, the `Follow conversation by email` button now appears in the Comments header.

When the link is tapped in the Notification body, it now compares the `post.crossPostMeta.postURL` with the tapped URL. If they are the same, the linked post is shown using the IDs from `post.crossPostMeta` (i.e. the original post). This allows the endpoint to be called that returns `subscribe` information for the original post, thus having the necessary information to display the follow button.

Bonus: fixed issues that were causing `Implicit conversion from enumeration type 'enum UIStackViewAlignment' to different enumeration type 'UIStackViewDistribution' (aka 'enum UIStackViewDistribution')` warnings in `ReaderPostHeaderView`.


To test:
- On a P2 you are following, ensure mobile notifications are enabled for new posts:

<kbd><img width="320" alt="notif_settings" src="https://user-images.githubusercontent.com/1816888/133518043-68f7130c-aa04-49b7-ab6b-c63f139982fd.png"></kbd>

- On a different P2, make a post cross posting to the P2 above.
- In the app, go to Notifications.
- Select the Notification for the xpost.
- In the body of the Notification, tap the link to the original post.
- Tap the Comments button at the bottom.
- Verify the `Follow(ing) conversation by email` button appears in the header and shows the correct state.

| Before | After |
|--------|-------|
| <img width="468" alt="Screen Shot 2021-09-15 at 11 38 39 AM" src="https://user-images.githubusercontent.com/1816888/133518962-ceb7f555-c00e-4c01-ba1a-814add676e92.png"> | <img width="480" alt="Screen Shot 2021-09-15 at 4 36 49 PM" src="https://user-images.githubusercontent.com/1816888/133518984-d377af74-1626-41ca-82c8-c6aad411d339.png"> |


## Regression Notes
1. Potential unintended areas of impact
N/A

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Tested different Notification types to ensure posts are loaded correctly.

3. What automated tests I added (or what prevented me from doing so)
N/A

PR submission checklist:
